contain-intrinsic-* missing properties (#21110)
mdn/content: files/en-us/web/css/contain-intrinsic-block-size/index.md

---
title: contain-intrinsic-block-size
slug: Web/CSS/contain-intrinsic-block-size
tags:
  - CSS
  - CSS Logical Property
  - CSS Property
  - Reference
  - experimental
  - contain-intrinsic-block-size
page-type: css-property
browser-compat: css.properties.contain-intrinsic-block-size
---

{{CSSRef}} {{SeeCompatTable}}

> The **`contain-intrinsic-block-size`** [CSS](/en-US/docs/Web/CSS) [logical property](/en-US/docs/Web/CSS/CSS_Logical_Properties) defines the block-size of an element that will be used for layout when it is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).
> 
> The block-size is the size of the element in the dimension perpendicular to the flow of text within a line.
> In a horizontal [writing mode](/en-US/docs/Web/CSS/writing-mode) like standard English this is the vertical dimension (height); for a vertical writing mode it is the horizontal dimension.

Reviewer: ```suggestion
The **`contain-intrinsic-block-size`** [CSS](/en-US/docs/Web/CSS) [logical property](/en-US/docs/Web/CSS/CSS_Logical_Properties) defines the block size of an element that a browser can use for layout when the element is subject to [size containment](/en-US/docs/Web/CSS/CSS_Containment#size_containment).

Block size is the size of an element in the dimension perpendicular to the flow of text within a line. In a horizontal [writing mode](/en-US/docs/Web/CSS/writing-mode) like standard English, block size is the vertical dimension (height); in a vertical writing mode, block size is the horizontal dimension.
```
- changed passive to active voice
- replaced the pronoun 'it' for clarity
please check suggestions for technical accuracy

## Syntax

```css
/* Keyword values */
contain-intrinsic-block-size: none;

/* <length> values */
contain-intrinsic-block-size: 1000px;
contain-intrinsic-block-size: 10rem;

/* auto <length> */
contain-intrinsic-block-size: auto 300px;

/* Global values */
contain-intrinsic-block-size: inherit;
contain-intrinsic-block-size: initial;
contain-intrinsic-block-size: revert;

> contain-intrinsic-block-size: unset;

Reviewer: should there be an entry for 'revert-layer'?

Author: @dipikabh Good question. Probably.

The short answer is that I cheated and just copied the data from contain-size.

Looking at the docs there is nothing in the `contain-*` that tells you that these apply so I looked at the ones that supposedly do and they say "can be applied to any CSS property". On that basis I will also added revert-layer and initial in all cases. If there are more cases than those, they are not marked up with the text in MDN.

Author: ```suggestion
contain-intrinsic-block-size: revert-layer;
contain-intrinsic-block-size: unset;
```

```

### Values

> The following values may be specified for an element.

Reviewer: ```suggestion
The following values can be specified for the intrinsic block size of an element:
```

Author: This is fine, but I don't see it as "better" because the context is completely clear from everything else about the page.  I generally prefer to say less unless extra words will help. 

That said, I accept comments unless I actively disagree and this is a matter of taste.

Reviewer: Sure, agree! Thank you :)

> - `none`
>   - : The element has no intrinsic block-size.
> - `<length>`
>   - : The element has the specified block-size ({{cssxref("&lt;length&gt;")}}).
> - `auto <length>`
>   - : A remembered value of the "normally rendered" element block-size if one exists and the element is skipping its contents (for example, when it is offscreen); otherwise the specified `<length>`.

Reviewer: ```suggestion
- `none`
  - : The element has no block size.
- `<length>`
  - : The element has the specified block size, expressed using the ({{cssxref("&lt;length&gt;")}}) data type.
- `auto <length>`
  - : When the element does not have any content (for example, when it is offscreen), the element takes on a remembered block size value of a "normally rendered" element. If a "normally rendered" element did not exist previously, the element has the specified `<length>` block size.
```

Author: Thanks, but I am not sure about this.
- There is a difference between intrinsic block size and block size, so we can't just remove "intrinsic" above.
- The changes you made for `auto` are wrong, but that doesn't mean than the original is optimal or right. What happens here is that if the element ever leaves size containment and is able to render its elements it does so, and the size is stored/"remembered". This is used only when the element is "skipping its contents", which is in effect when it is offscreen and not about to enter the viewport. I have tried tidy this below. Feel free to suggest again.

```suggestion
- `none`
  - : The element has no intrinsic block size.
- `<length>`
  - : The element has the specified block size, expressed using the ({{cssxref("&lt;length&gt;")}}) data type.
- `auto <length>`
  - : When the element is in size containment and skipping its contents (for example, when it is offscreen and `content-visibility: auto` is set) the block size used is remembered from the actual size of the element when it was last able to render its child elements. If the element has has never rendered its child elements and has no remembered value, or if it is not skipping its contents, the block size is the specified `<length>`.
```

Reviewer: This is definitely better, thank you!

## Description

The property is commonly applied alongside elements that can trigger size containment, such as [`contain: size`](/en-US/docs/Web/CSS/contain) and [`content-visibility`](/en-US/docs/Web/CSS/content-visibility).

> Size containment allows a user agent to layout an element as though it had a fixed size, preventing unnecessary reflows by avoiding the re-rendering of child elements to determine the actual size (thereby improving user experience).

Reviewer 2: ```suggestion
Size containment allows a user agent to lay out an element as though it had a fixed size, preventing unnecessary reflows by avoiding the re-rendering of child elements to determine the actual size (thereby improving user experience).
```

Author: Fixed by another one.

> Size containment allows a user agent to layout an element as though it had a fixed size, preventing unnecessary reflows by avoiding the re-rendering of child elements to determine the actual size (thereby improving user experience).

Reviewer: ```suggestion
Size containment allows a user agent to lay out an element as though it had a fixed size. This prevents unnecessary reflows by avoiding the re-rendering of child elements to determine the actual size (thereby improving user experience).
```
just splitting the long sentence

> By default, size containment treats elements as though they had no contents, and may collapse the layout in the same way as if the contents had no width or height.
> The `contain-intrinsic-block-size` property allows authors to specify an appropriate value to be used as the block-size for layout.
> 
> The `auto <length>` value allows the block-size of the element to be stored if the element is ever "normally rendered" (with its child elements), and then used instead of the specified value when the element is skipping its contents.
> This allows offscreen elements with [`content-visibility: auto`](/en-US/docs/Web/CSS/content-visibility) to benefit from size containment without developers having to be as precise in their estimates of element size.
> The remembered value is not used if the child elements are being rendered (if size containment is enabled, the `<length>` will be used).

Reviewer: ```suggestion
By default, size containment treats elements as though they had no contents and may collapse the layout in the same way as if the contents had no width or height.
The `contain-intrinsic-block-size` property allows authors to specify an appropriate value to be used as the block-size for layout.

The `auto <length>` value allows the block-size of an element to be stored if the element is ever "normally rendered" (with its child elements) and then used instead of the specified value when the element does not have any content.
This allows offscreen elements with [`content-visibility: auto`](/en-US/docs/Web/CSS/content-visibility) to benefit from size containment without developers having to be precise in their estimates of element size.
The remembered value is not used if the child elements are being rendered; if size containment is enabled, the `<length>` value will be used.
```

> 

Reviewer: Can we have an "Examples" section?
